types/example.pp

```diff
@@ -1,14 +1,14 @@
 type Patterndb::Example = Struct[
   {
-    program      => String[1],
+    program      => String,
     test_message => String[1],
     test_values  => Optional[
       Hash[
         String[1],
         Struct[
           {
-            type  => Optional[Enum['datetime', 'double', 'integer', 'json', 'string']],
-            value => String[1],
+            type  => Optional[Enum['datetime', 'double', 'integer', 'json', 'string', 'null']],
```

Comment:
The XML schema does not list the allowed values so this list was the one from the comment. 'null' does not appear there, but indeed, when a field is absent, the test match its current value as `` and its current type as `null`, so I guess it make sense to add it.

While changing it, I think we should change the `Optional` location which is not exactly what I expected: with this syntax we allow an undef for "type", but in reality, "type" is optional and if set should be one of the expected values:

```puppet
type Foo = Struct[
  {
    foo => Optional[String[1]],
  },
]

warning({} =~ Foo)               # true
warning({foo => ""} =~ Foo)      # false
warning({foo => "value"} =~ Foo) # true
warning({foo => undef } =~ Foo)  # true <---

type Bar = Struct[
  {
    Optional[foo] => String[1],
  },
]

warning({} =~ Bar)               # true
warning({foo => ""} =~ Bar)      # false
warning({foo => "value"} =~ Bar) # true
warning({foo => undef } =~ Bar)  # false <---

```

So I suggest:
```suggestion
            Optional[type]  => Enum['datetime', 'double', 'integer', 'json', 'string', 'null'],
```

Reply:
Wait, `warning({} =~ Foo)               # true` ? Isn't that a bug ?